core/util/src/main/java/edu/cmu/cs/dennisc/codec/BufferUtilities.java

```diff
@@ -190,69 +190,81 @@ public class BufferUtilities {
   private BufferUtilities() {
   }
 
-  private static void encodeHeader(BinaryEncoder encoder, Buffer buffer, int bitsPerPrimitive, boolean isDirect, boolean isNativeRequired) {
+  private static void encodeHeader(BinaryEncoder encoder, Buffer buffer, int bitsPerPrimitive) {
+    encodeHeader(encoder, buffer, bitsPerPrimitive, true);
+  }
+
+  private static void encodeHeader(BinaryEncoder encoder, Buffer buffer, int bitsPerPrimitive, boolean isNativeRequired) {
     buffer.rewind();
-    BufferDetails bufferDetails = new BufferDetails(buffer, bitsPerPrimitive, isDirect, isNativeRequired);
+    BufferDetails bufferDetails = new BufferDetails(buffer, bitsPerPrimitive, buffer.isDirect(), isNativeRequired);
     bufferDetails.encodeHeader(encoder);
   }
 
-  public static void encode(BinaryEncoder encoder, ByteBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Character.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, ByteBuffer buffer) {
+    encodeHeader(encoder, buffer, Character.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, CharBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Character.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, CharBuffer buffer) {
+    encodeHeader(encoder, buffer, Character.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, ShortBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Short.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, ShortBuffer buffer) {
+    encodeHeader(encoder, buffer, Short.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
```

Comment:
can you add a comment here about which one of these to choose for what purpose, for the poor lost soul who stumbles in here later?

Reply:
It looks like which to call is based on what kind of buffer you have.
All the methods exist only to service `AbstractBinaryEncoder.encodeObject()` and its type based testing, but is any of that a useful comment in the code?

Reply:
I do not know how the buffer utilities authors made these choices. Understanding and documenting this seems unrelated to the point of this PR. Can I just say I don't know and move on?